### ISSUE TITLE
Error: Failed timestamp transformation

### ISSUE THREAD

**Issue by nicolasfostier** [googleapis/python-aiplatform#1848]
Hello,

I encounter nearly the same issue than [https://github.com/googleapis/python-aiplatform/issues/1034](https://github.com/googleapis/python-aiplatform/issues/1034), except that for me it's throught the endpoint prediction, not the batch prediction. I've tried multiple datetime formats, none is working.

#### Environment details

  - OS type and version: Ubuntu 20.04
  - Python version: 3.8.10
  - pip version: 20.0.2
  - `google-cloud-aiplatform` version: 1.19.1

#### Steps to reproduce

  1. Train an AutoML model with a column that is using the timestamp transformation.
  2. Deploy the trained model on a endpoint
  3. Try to perform a prediction with the endpoint

#### Stack trace
```
  File "venv/lib/python3.8/site-packages/google/cloud/aiplatform/models.py", line 1503, in predict
    prediction_response = self._prediction_client.predict(
  File "venv/lib/python3.8/site-packages/google/cloud/aiplatform_v1/services/prediction_service/client.py", line 589, in predict
    response = rpc(
  File "venv/lib/python3.8/site-packages/google/api_core/gapic_v1/method.py", line 154, in __call__
    return wrapped_func(*args, **kwargs)
  File "venv/lib/python3.8/site-packages/google/api_core/grpc_helpers.py", line 74, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.InvalidArgument: 400 {"error": "Column: creationDate. Error: Failed timestamp transformation."}
```

**Comment by nicolasfostier** [comment]
After more attempts, i've successfully get a prediction with the format : "2009-07-12 11:03:12+00:00". It seems weird to me that this documentation page [https://cloud.google.com/vertex-ai/docs/datasets/data-types-tabular#timestamp](https://cloud.google.com/vertex-ai/docs/datasets/data-types-tabular#timestamp) is listing a lot of format for training but when doing endpoint prediction, the working format is not even listed in this page.

Moreover, i think it will be great if the error message could indicate the expected datetime format.

**Comment by NicholasJallan** [comment]
Hello !
I'm experiencing the exact same problem, but currently i can't get a valid prediction. What's very distiurbing is that i can get predictions with a json sent to the endpoint published on Vertex, but not on the very same model deployed locally with any of the recent images given here https://console.cloud.google.com/gcr/images/cloud-automl-tables-public/global/model_server
I guess there is a slight difference somewhere between the image used on my account and any of those we can instantiate locally.
I'm looking for an answer too :-)

**Comment by rosiezou** [comment]
@NicholasJallan @nicolasfostier Thanks for raising this issue. I will try to repro the same error message this week and get back to you soon.

**Comment by rosiezou** [comment]
Hi @NicholasJallan and @nicolasfostier. So sorry for the delay!! Lots of things came up during the past few weeks and I was swamped. I finally got around to take a look at this issue and there are multiple aspects to the issues that you are both seeing. 

1. The root cause for the prediction failure appears to be a discrepancy between the valid data input format accepted for AutoML model training and the valid data input format accepted in the prediction function. The `predict` function in `models.py` doesn't do any data validation and is only a thin wrapper that invokes the gRPC call via the prediction client.
2. Documentation should be aligned across all parts of the code base.
3. The public images that are uploaded might need more thorough validation and testing.

I will work with our cross-functional teammates to resolve the above-mentioned items in the following way:

1. Adding input data validation in the frontend SDK might not be the best solution in this case. It will create long-term tech debt for code maintenance and testing since the backend is implemented separately from the SDK. I will create a separate ticket and forward it to our backend team so that they could add more actionable error messages and/or add input data validation/conversion features as they see fit.
2. I will ping the tech writing team so that the documentation can be more helpful.
3. There is a separate team who owns the uploaded public images. I will reach out to them and ask them to take a look at the issue.

I'll keep this issue open and update the progress here. @NicholasJallan, it looks like @nicolasfostier's solution will resolve the prediction errors that you're encountering. So I would recommend that as a workaround solution while we wait on a more permanent resolution.

**Comment by TheMichaelHu** [comment]
Hey all, can you check if the time format used for prediction is the same as the one used for training? If that is the issue, I will update this error message to be more helpful. If not, we may not be storing the training time format correctly.

**Comment by rosiezou** [comment]
Closing this due to inactivity. @NicholasJallan @nicolasfostier please re-open if you're still encountering the same issues.

**Comment by NicholasJallan** [comment]
This problem is unresolved, keep it opened until solved.

**Comment by NicholasJallan** [comment]
> Hey all, can you check if the time format used for prediction is the same as the one used for training? If that is the issue, I will update this error message to be more helpful. If not, can you share your project number and model id so we can take a closer look? You can email [cloud-automl-tables-feedback@google.com](mailto:cloud-automl-tables-feedback@google.com) if you don't want to share that information on a public GitHub issue.

It  was the same format that was used. It's working when the model is directly exposed through gcp hosting, but doesn't work when the model is dockerised and deployed locally.
